lib/Packager/module-resolver.es6

```diff
@@ -375,8 +375,8 @@ function EXPAND (target, log, cache, opts, cwd) {
 		if(result.json) {
 			let moduleDir = result.json.moduleDir;
 			if (!moduleDir) {
-				let libDir = path.join(opts.cwd, opts.libDir, dir);
-				if (libDir === step) {
+				let libDir = path.resolve(step, '..')
```

Comment:
The idea is to check if the current expansion step directory is a library top directory (a directory in the opts.path (which includes the resolved libDir).

It was established previously that libraries package.json (ex./ .../lib/garnet/package.json) should default to moduleDir:"src"  property. This chunk of code, executed on a package.json entry discovered without a moduleDir property, checks the current step's parent (.../lib from ../lib/garnet) and then checks if that is a valid opts.path. That way, if it is indeed a library, the library package.json can have moduleDir set as a default 'src' subdirectory and the modules can be expanded properly.

What case would this not work for?